src/pages/Account/AdminEvent.tsx

```diff
@@ -61,26 +47,27 @@ const AdminEvent: React.FunctionComponent<PropsType> = ({
   onHide,
   venueId,
 }) => {
-  const { register, handleSubmit, errors, formState } = useForm<
-    CreateEventFormData
-  >({
+  const { register, handleSubmit, errors, formState } = useForm<EventInput>({
     mode: "onChange",
     validationSchema,
   });
-  const onSubmit = async (data: CreateEventFormData) => {
-    const start = dayjs(`${data.start_date} ${data.start_time}`);
-    const end = dayjs(`${data.end_date} ${data.end_time}`);
-    const event: Omit<VenueEvent, "id"> = {
-      name: data.name,
-      description: data.description,
-      start_utc_seconds: start.unix(),
-      duration_minutes: end.diff(start, "minute"),
-      price: Math.floor(data.price * 100),
-      collective_price: 0,
-    };
-    await postEvent(venueId, event);
-    onHide();
-  };
+  const onSubmit = useCallback(
+    async (data: EventInput) => {
+      const start = dayjs(`${data.start_date} ${data.start_time}`);
+      const end = dayjs(`${data.end_date} ${data.end_time}`);
+      const event: Omit<VenueEvent, "id"> = {
```

Comment:
event: EventInput

Reply:
They have very different schema (EventInput has start_date, start_time, end... when VenueEvent only has start_seconds and duration).

Difference is I do the transformation at the form level when you did it here. I'd rather not expose the internal representation the Form needs Any thoughts?